pypeit/spectrographs/spectrograph.py

```diff
@@ -631,6 +672,27 @@ def get_meta_value(self, inp, meta_key, required=False, ignore_bad_header=False,
 
         return retvalue
 
+    def set_detector_par(self, par, det, value, force_update=False):
```

Comment:
We very much prefer not to update any `par` values after the code starts running.
Can you explain why this method is needed?

Reply:
I tend to agree with that philosophy, but (it seems for the sake of time) the default KCWI calibrations scripts take different calibrations with a different number of amplifier modes. So, the `oscansec`, `datasec`, `numamplifiers`, `ronoise`, `gain` parameters all depend on the amplifier setup. These parameters therefore have to be modified during the reduction of each frame... not ideal, I know, but I couldn't think of any other way around it. I'm also guessing that many folks will use the "default" calibrations script.

The best that can be done (I think) is to pull the information from the headers...

Reply:
ok, let's discuss a bit on Thursday.

Reply:
As discussed by telecon, we'll strip this for now and create a Detector class (I think that's what was decided!)